src/main/java/net/logstash/logback/util/ReusableJsonFormatterPool.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.logstash.logback.util;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.lang.ref.SoftReference;
+import java.util.Deque;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import net.logstash.logback.composite.CompositeJsonFormatter;
+
+import ch.qos.logback.core.spi.DeferredProcessingAware;
+
+/**
+ * @author brenuart
+ *
+ */
+public class ReusableJsonFormatterPool<Event extends DeferredProcessingAware> {
```

Comment:
Would love some documentation on this.

What is it for?

Is it intended to be used outside of logstash-logback-encoder?

Is it threadsafe?